### PR TITLE
[3h+] issue #1493 - rm reply headers from url

### PR DESCRIPTION
**Pull request by michael-volynets**
2h 30m so far

### CODE REVIEW

**Comment by michael-volynets** [comment]
+ 20m

File: extension/chrome/elements/compose.ts [hunk 1]
```diff
@@ -64,43 +57,47 @@ Catch.try(async () => {
...
+  let to: string[] = [];
+  let cc: string[] = [];
+  let bcc: string[] = [];
+  if (threadId) {
+    const fetchSuccess = await (async () => {
+      Xss.sanitizePrepend('#new_message', Ui.e('div', { id: 'loader', html: 'Loading secure reply box..' + Ui.spinner('green') }));
+      try {
+        const thread = await Google.gmail.threadGet(acctEmail, threadId, 'metadata');
+        const gmailMsg = await Google.gmail.msgGet(acctEmail, thread.messages[thread.messages.length - 1].id, 'metadata');
+        const reply = Google.determineReplyCorrespondents(acctEmail, storage.addresses || [], gmailMsg);
```
Comment (tomholub) [inline extension/chrome/elements/compose.ts]: If I'm reading this correctly, you could simplify as:
```js
const {from, to, cc, bcc} = Google.determineReplyCorrespondents(acctEmail, storage.addresses || [], gmailMsg);
```

**Comment by tomholub** [comment]
You can ignore the codacy check

File: extension/chrome/elements/compose.ts (hunk 1 above)
Comment (michael-volynets) [reply extension/chrome/elements/compose.ts]: I couldn't make it exactly the same way you wrote but I have made it a little better

**Comment by michael-volynets** [comment]
> I feel like there should be some more code in webmail_elements_replacer (or similar) to remove? Like parsing recipients and subject from the DOM, and adding that to url params, etc.

I removed parsing `subject` form that file.
I think I've mostly done it in some of previous PRs because I can't see parsing of a message in `gmail_element_replacement`, it parses only ThreadMSgId, and threadId now.

**Comment by tomholub** [comment]
> > I feel like there should be some more code in webmail_elements_replacer (or similar) to remove? Like parsing recipients and subject from the DOM, and adding that to url params, etc.
> 
> I removed parsing `subject` form that file.
> I think I've mostly done it in some of previous PRs because I can't see parsing of a message in `gmail_element_replacement`, it parses only ThreadMSgId, and threadId now.

Sounds good, as long as we no longer pass subject, to, ..., ... in the url either? And the XssSafeFactory was cleaned up of this as well?

**Pull request merged**